### PR TITLE
CORE-37060 Added setUserXdm and setUserData methods to event.

### PR DESCRIPTION
**Pull request by Aaronius**
## Description
Removed applyUserProvidedData and instead added `setUserXdm` and `setUserData` methods to the event. The event, upon serialization, will take care to override Alloy-produced data with user-provided data.

## Related Issue
https://jira.corp.adobe.com/browse/CORE-37060
https://jira.corp.adobe.com/browse/CORE-36755

## Motivation and Context
Easier for engineers to understand.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [x] I have submitted a [documentation](https://github.com/AdobeDocs/alloy-docs) pull request or no changes are needed.
- [x] I have made any necessary test changes and all tests pass.
- [x] I have run the Sandbox successfully.

### CODE REVIEW

File: src/core/createEventManager.js [hunk 1]
```diff
@@ -10,15 +10,27 @@ OF ANY KIND, either express or implied. See the License for the specific languag
...
+     * the request payload.
+     * @param {Object} [options]
+     * @param {boolean} [options.isViewStart=false] Whether the event is a
+     * result of the start of a view. This will be passed to components
+     * so they can take appropriate action.
+     * @returns {*}
+     */
     sendEvent(event, options = {}) {
-      const { isViewStart = false, applyUserProvidedData = noop } = options;
+      const { isViewStart = false } = options;
```
Comment (Aaronius) [inline src/core/createEventManager.js]: It seems odd that we pass `isViewStart` around separately from `event`, but we manage `isDocumentUnloading` and `expectsResponse` directly on the event object itself. Does anyone else think we should move `isViewStart` into the event itself?

Comment (jfkhoury) [reply src/core/createEventManager.js]: I agree, good catch.

Comment (artur-ciocanu) [reply src/core/createEventManager.js]: +1

Comment (artur-ciocanu) [reply src/core/createEventManager.js]: @Aaronius we have `isViewStart` and `isDocumentUnloading()` and then `expectsResponse()` should we rename it to `isExpectingResponse()`? Thoughts?

Comment (Aaronius) [reply src/core/createEventManager.js]: Yeah, probably. I'll make a separate issue to track this.

Comment (Aaronius) [reply src/core/createEventManager.js]: For reference: https://jira.corp.adobe.com/browse/CORE-37113

**Pull request merged**